[JSC][Win][Clang 20] LinkBuffer.h: warning: private field 'm_ownerUID' is not used [-Wunused-private-field]

fujii opened the pull request:
#### e48690b09468f9a19bbc3aeeb5d2c38a3d279984
<pre>
[JSC][Win][Clang 20] LinkBuffer.h: warning: private field &apos;m_ownerUID&apos; is not used [-Wunused-private-field]
<a href="https://bugs.webkit.org/show_bug.cgi?id=290355">https://bugs.webkit.org/show_bug.cgi?id=290355</a>

Reviewed by Yusuke Suzuki.

clang-cl v20.1 emitted unused-private-field warnings for m_ownerUID
and m_isRewriting of LinkBuffer class for Windows port. Those fields
were only used for OS(LINUX) || OS(DARWIN) because PerfLog class
supported only the platforms. PerfLog supports only ELF format.

This problem could be worked around by conditioning out the fields in
LinkBuffer class for Windows. However, it will introduce more #if
directives in the class.

Rather than doing that, this change makes PerfLog class to be able to
compile for Windows with less platform-specific #if directives. This
change doesn&apos;t make PerfLog class usable nor enable
JSC_logJITCodeForPerf option for Windows. Options::isAvailable enables
it only for OS(LINUX) || OS(DARWIN) at the moment.

* Source/JavaScriptCore/assembler/LinkBuffer.cpp:
(JSC::LinkBuffer::logJITCodeForPerf):
* Source/JavaScriptCore/assembler/PerfLog.cpp:
(JSC::getCurrentThreadID):
(JSC::PerfLog::PerfLog):
* Source/JavaScriptCore/assembler/PerfLog.h:

Canonical link: <a href="https://commits.webkit.org/292740@main">https://commits.webkit.org/292740@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cb12dcc986dc912e6fa266e9654d7557994b9faa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96979 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16593 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6793 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102055 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/47500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16889 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25042 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/73875 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/47500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99982 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/12730 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/87716 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54210 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/12487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5551 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/46827 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/89652 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/82569 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/5632 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104076 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/95600 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24047 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/17551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/82923 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/24300 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83803 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82318 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26998 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/4545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/17550 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15647 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24009 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/29164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/119227 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/23836 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/33501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/27148 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25409 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/JavaScriptCore/assembler/PerfLog.cpp:
```diff
@@ -177,6 +181,9 @@ PerfLog::PerfLog()
         m_file = fdopen(m_fd, "wb");
+#else
+        m_file = fopen(filename.toCString().data(), "wb+");
+#endif
```
How about using `_fdopen` for Windows instead and keeping the other code as is too on Windows?

Constellation commented on Source/JavaScriptCore/assembler/PerfLog.cpp:
```diff
@@ -166,6 +169,7 @@ PerfLog::PerfLog()
             filename.print("/tmp");
         filename.print("/jit-", getCurrentProcessID(), ".dump");
+#if OS(UNIX)
         m_fd = open(filename.toCString().data(), O_CREAT | O_TRUNC | O_RDWR, 0666);
```
Using `_open` for Windows.

Constellation commented on Source/JavaScriptCore/assembler/PerfLog.h:
```diff
@@ -56,10 +56,12 @@ class PerfLog {
     FILE* m_file { nullptr };
     void* m_marker { nullptr };
     uint64_t m_codeIndex { 0 };
+#if OS(UNIX)
```
Then we do not need this guard.

fujii replied:
Fixed.

Constellation commented on Source/JavaScriptCore/assembler/PerfLog.cpp:
```diff
@@ -27,24 +27,41 @@
+    return _open(filename, oflag, pmode);
+}
+
+inline FILE* fdopen(int fd, const char* mode)
```
Let's put static

fujii replied:
Fixed.

Constellation commented on Source/JavaScriptCore/assembler/PerfLog.cpp:
```diff
@@ -27,24 +27,41 @@
+#if OS(WINDOWS)
+#include <io.h>
+
+inline int open(const char* filename, int oflag, int pmode)
```
Ditto

fujii committed:
[JSC][Win][Clang 20] LinkBuffer.h: warning: private field 'm_ownerUID' is not used [-Wunused-private-field]
https://bugs.webkit.org/show_bug.cgi?id=290355

Reviewed by Yusuke Suzuki.

clang-cl v20.1 emitted unused-private-field warnings for m_ownerUID
and m_isRewriting of LinkBuffer class for Windows port. Those fields
were only used for OS(LINUX) || OS(DARWIN) because PerfLog class
supported only the platforms. PerfLog supports only ELF format.

This problem could be worked around by conditioning out the fields in
LinkBuffer class for Windows. However, it will introduce more #if
directives in the class.

Rather than doing that, this change makes PerfLog class to be able to
compile for Windows with less platform-specific #if directives. This
change doesn't make PerfLog class usable nor enable
JSC_logJITCodeForPerf option for Windows. Options::isAvailable enables
it only for OS(LINUX) || OS(DARWIN) at the moment.

* Source/JavaScriptCore/assembler/LinkBuffer.cpp:
(JSC::LinkBuffer::logJITCodeForPerf):
* Source/JavaScriptCore/assembler/PerfLog.cpp:
(JSC::getCurrentThreadID):
(JSC::PerfLog::PerfLog):
* Source/JavaScriptCore/assembler/PerfLog.h:

Canonical link: https://commits.webkit.org/292740@main
